dojo/tools/sonarqube_api/importer.py

```diff
@@ -143,6 +154,84 @@ def import_issues(self, test):
+                component = client.get_project(config.sonarqube_project_key)
+            else:  # https://github.com/DefectDojo/django-DefectDojo/pull/4676 cases no. 2, 4 and 7
+                component = client.find_project(test.engagement.product.name)
+
+            hotspots = client.find_hotspots(component['key'])
+            logging.info('Found {} hotspots for project {}'.format(len(hotspots), component["key"]))
+
+            for hotspot in hotspots:
+                status = hotspot['status']
+
+                if self.is_reviewed(status):
+                    continue
+
+                type = 'SECURITY_HOTSPOT'
+                if len(hotspot['message']) > 513:
+                    title = textwrap.shorten(text=hotspot['message'], width=507)
+                else:
+                    title = hotspot['message']
+                component_key = hotspot['component']
+                line = hotspot.get('line')
+                rule_id = hotspot['key']
+                rule = client.get_hotspot_rule(rule_id)
+                severity = self.convert_sonar_review_priority(hotspot['vulnerabilityProbability'])
+                description = self.clean_rule_description_html(rule['vulnerabilityDescription'])
+                cwe = self.clean_cwe(rule['riskDescription'])
+                references = self.get_references(rule['riskDescription'])
+
+                sonarqube_issue, _ = Sonarqube_Issue.objects.update_or_create(
+                    key=hotspot['key'],
+                    defaults={
+                        'status': status,
+                        'type': type
+                    }
+                )
+
+                # Only assign the SonarQube_issue to the first finding related to the issue
+                if Finding.objects.filter(sonarqube_issue=sonarqube_issue).exists():
+                    sonarqube_issue = None
+
+                find = Finding(
+                    title=title,
+                    cwe=cwe,
+                    description=description,
+                    test=test,
+                    severity=severity,
+                    references=references,
+                    file_path=component_key,
+                    line=line,
+                    active=True,
+                    verified=self.is_confirmed(status),
+                    false_p=False,
+                    duplicate=False,
+                    out_of_scope=False,
+                    static_finding=True,
+                    sonarqube_issue=sonarqube_issue,
+                )
+                items.append(find)
+
+        except Exception as e:
+            logger.exception(e)
+            create_notification(
+                event='other',
+                title='SonarQube API hotspot import issue',
+                description=e,
+                icon='exclamation-triangle',
+                source='SonarQube API'
+            )
```

Comment:
1. Don't send notification.
2. The exception is swallowed. It will return empty list that will close existing findings with some use cases.
You can remove the try/catch

Reply:
Done